### PR TITLE
Update AnalyzerReleases file to reflect new analyzers for .NET9 release

### PR DESCRIPTION
**Pull request by buyaa-n**
Update AnalyzerReleases file to reflect new analyzers for .NET9 release same as in https://github.com/dotnet/roslyn-analyzers/pull/6940

This need to be ported to release/9.0.1xx branch.
The new analyzer added in 10.0 will be removed from that port

CC @carlossanlop @jeffhandley

### CODE REVIEW

File: src/NetAnalyzers/Core/AnalyzerReleases.Unshipped.md [hunk 1]
```diff
@@ -4,13 +4,4 @@
...
-CA2022 | Reliability | Warning | AvoidUnreliableStreamReadAnalyzer, [Documentation](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2022)
-CA2023 | Reliability | Warning | LoggerMessageDefineAnalyzer, [Documentation](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2023)
-CA2262 | Usage | Info | ProvideHttpClientHandlerMaxResponseHeaderLengthValueCorrectly, [Documentation](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2262)
-CA2263 | Usage | Info | PreferGenericOverloadsAnalyzer, [Documentation](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2263)
-CA2264 | Usage | Warning | DoNotPassNonNullableValueToArgumentNullExceptionThrowIfNull, [Documentation](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2264)
-CA2265 | Usage | Warning | DoNotCompareSpanToNullAnalyzer, [Documentation](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2265)
+CA2023 | Reliability | Warning | LoggerMessageDefineAnalyzer, [Documentation](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2023)
```
Comment (carlossanlop) [inline src/NetAnalyzers/Core/AnalyzerReleases.Unshipped.md:7]: Is this CA2023 what you meant in the description?:

> The new analyzer added in 10.0 will be removed from that port

**Review by carlossanlop — APPROVED** [review]
/backport to release/9.0.1xx

File: src/NetAnalyzers/Core/AnalyzerReleases.Unshipped.md (hunk 1 above)
Comment (buyaa-n) [reply src/NetAnalyzers/Core/AnalyzerReleases.Unshipped.md:7]: Yes

**Pull request merged**